src/App.js

```diff
@@ -11,39 +13,263 @@ class App extends React.Component {
+      }));
+    }
+  };
+
+  // function to restart entire game
+  restartGame = () => {
     this.setState({
-      currCards: newCurrCards,
+      cardDeck: makeShuffledDeck(),
+      // currCards holds the cards from the current round
+      currCards: [],
+      player1Score: 0,
+      player2Score: 0,
+      currWinner: null,
+      hasGameBegin: false,
+      numMatchesPlayer1Won: 0,
+      numMatchesPlayer2Won: 0,
+      matchNumber: 1,
     });
   };
 
-  render() {
-    // You can write JavaScript here, just don't try and set your state!
+  // function to start new match
+  startNewMatch = () => {
+    this.setState((prevState) => ({
+      cardDeck: makeShuffledDeck(),
+      // currCards holds the cards from the current round
+      currCards: [],
+      player1Score: 0,
+      player2Score: 0,
+      currWinner: null,
+      hasGameBegin: false,
+      matchNumber: prevState.matchNumber + 1,
+    }));
+  };
 
-    // You can access your current components state here, as indicated below
-    const currCardElems = this.state.currCards.map(({ name, suit }) => (
+  render() {
+    let { currCards, currWinner, player1Score, player2Score } = this.state;
+    let currCardElems = currCards.map(({ name, suit }) => (
       // Give each list element a unique key
-      <div key={`${name}${suit}`}>
-        {name} of {suit}
-      </div>
+      <PlayingCards key={`${name}${suit}`} name={name} suit={suit} />
     ));
+    // You can write JavaScript here, just don't try and set your state!
+    let winnerPerRound = `Player ${currWinner} won this round!`;
+    // You can access your current components state here, as indicated below
+
+    let winnerPerMatch;
+    if (player1Score > player2Score) {
+      winnerPerMatch = (
+        <>
+          <p>
+            Game over! <br />
+            Player 1 won this match. Would you like to continue?
+          </p>
+        </>
+      );
+    } else if (player1Score < player2Score) {
+      winnerPerMatch = (
+        <>
+          <p>
+            Game over!
+            <br /> Player 2 won this match. Would you like to continue?
+          </p>
+        </>
+      );
+    } else {
+      winnerPerMatch = (
+        <>
+          <p>
+            Game over! <br />
+            Both tie. Would you like to continue?
+          </p>
+        </>
+      );
+    }
+
+    const dealCardsButton = (
+      <>
+        <Button variant="contained" color="secondary" onClick={this.dealCards}>
+          Deal
+        </Button>
+      </>
+    );
+
+    const continueGameButton = (
+      <>
+        <Button
+          variant="contained"
+          color="success"
+          onClick={this.startNewMatch}
+        >
+          Yes, Enter Game
+        </Button>
+      </>
+    );
+
+    const exitGameButton = (
+      <>
+        <Button variant="contained" color="error" onClick={this.restartGame}>
+          No, Exit game
+        </Button>
+      </>
+    );
```

Comment:
Component, and component :) Since we got a few button components, maybe we can make a generic button component?
Maybe used like so:
`<Button variant={} color={} onClick={} text={} />`